dev-docs/RFCs/v6.x/lighting-rfc.md

```diff
@@ -0,0 +1,51 @@
+# RFC: Lighting
+
+* **Authors**: Jian Huang
+* **Date**: Sep. 2018
+* **Status**: For Review
+
+## Motivation
+
+Lighting is an important rendering effects in deck.gl, the current lighting module has some issues.
+* Light source is duplicated in each layer
+* Only support point light source
+* Light source and material parameters are bundled together
+* Specular highlight is tinted by object color
+
+## Goals
+* Light sources are owned by scene and shared by layers
+* Support ambient, point and directional light sources
+* Separate light source and material parameters
+* Specular highlight is not tinted by object color
+
+## Proposal
+
+### Light Source
+Three light source classes specify light source details for the scene
+* Ambient light
+color (vec3): color of ambient light source
+intensity (float): strenght of ambient light source
+* Point light
```

Comment:
Only one point light?

Only one directional light?

Reply:
here is to specify the props for each light source class, the number is not restricted

Reply:
> the number is not restricted

Ok, as long as we can have more than one of each, just 2-3 should be OK.

Somehow the number may need to be restricted because shaders tend to need static iteration limits and length limits on array uniforms.

So it can make sense to choose a modest limit. It can of course be decided in some constant.  It could also be handled by `defines` in the shader module, if we really wanted to keep it variable.